src/lib/lessons.ts

```diff
@@ -1,116 +1,149 @@
-        hls_url
-        dash_url
-        http_url
-        media_url
-        lesson_view_url
-        thumb_url
-        icon_url
-        download_url
-        staff_notes_url
-        repo_url
-        code_url
-        created_at
-        updated_at
-        published_at
-        collection {
-          ... on Playlist {
-            title
+const lessonQuery = groq`
+*[_type == 'lesson' && slug == $slug][0]{
+  title,
+  slug,
+  description
+}`
+
+/**
+ * loads LESSON METADATA from Sanity
+ * @param slug
+ */
+async function loadLessonMetadataFromSanity(slug: string) {
+  const params = {
+    slug,
+  }
+
+  const lesson = await sanityClient.fetch(lessonQuery, params)
+
+  return lesson
+}
+
+export async function loadLesson(
+  slug: string,
+  token?: string,
+): Promise<LessonResource> {
+  token = token || getAccessTokenFromCookie()
+  const graphQLClient = getGraphQLClient(token)
+
+  const variables = {
+    slug: slug,
+  }
+
+  const {lesson} = await graphQLClient.request(
+    loadLessonGraphQLQuery,
+    variables,
+  )
+
+  const lessonMetadataFromSanity = await loadLessonMetadataFromSanity(slug)
+
+  const lessonMetadata = {...lesson, ...lessonMetadataFromSanity}
+
+  // comments are user-generated content that must come from the egghead-rails
+  // backend, so load those separately from the rest of lesson metadata.
+  const comments = await loadLessonComments(slug, token)
+
+  return {...lessonMetadata, comments} as LessonResource
+}
+
+const loadLessonGraphQLQuery = /* GraphQL */ `
+  query getLesson($slug: String!) {
+    lesson(slug: $slug) {
+      slug
+      title
+      description
+      duration
+      next_up_url
+      free_forever
+      path
+      transcript
+      transcript_url
+      subtitles_url
+      hls_url
+      dash_url
+      http_url
+      media_url
+      lesson_view_url
+      thumb_url
+      icon_url
+      download_url
+      staff_notes_url
+      repo_url
+      code_url
+      created_at
+      updated_at
+      published_at
+      collection {
+        ... on Playlist {
+          title
+          slug
+          type
+          square_cover_480_url
+          path
+          lessons {
             slug
             type
-            square_cover_480_url
             path
-            lessons {
-              slug
-              type
-              path
-              title
-              completed
-              duration
-              thumb_url
-              media_url
-            }
-          }
-          ... on Course {
             title
-            slug
-            type
-            square_cover_480_url
-            path
-            lessons {
-              slug
-              type
-              path
-              title
-              completed
-              duration
-              thumb_url
-              media_url
-            }
+            completed
+            duration
+            thumb_url
+            media_url
           }
         }
-        course {
```

Comment:
remove `course` because it wasn't being used anywhere downstream once it was loaded in.

<img width="728" alt="CleanShot 2022-07-26 at 15 02 10@2x" src="https://user-images.githubusercontent.com/694063/181101720-05d24f2b-0ad1-49e9-9beb-34ced5b17cb7.png">


Commit: https://github.com/eggheadio/egghead-next/pull/1148/commits/6f3bdd19213606cd3e312cf2662fd4cbadd491d1